### PR TITLE
Added code to display templates on the dashboard screen

### PR DESCRIPTION
**Pull request by bryanansong**
## Description
Added fetch method to query all the user's templates. I made routes to pass this data to my workoutTemplatesController file where I have a method to create a WorkoutTemplate record along with an Exercise and a WorkoutTemplateExercise record. 
<what this pull request is doing>
<what will be done in later PRs and not included here>

## Test Plan

https://github.com/bryanansong/ZenFitness/assets/50881304/e3c92658-896b-41b8-8dc4-e87458fd8670

<insert images or gifs of feature>
<otherwise, say how code was tested if not UI facing>
<any edge cases you might have specifically tested>

### CODE REVIEW

File: frontend/src/components/CreateTemplateCard/CreateTemplateCard.tsx [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
+
+const CreateTemplateCard: React.FC<CreateTemplateCardProps> = ({ fetchTemplates }) => {
   const [isModalOpen, setIsModalOpen] = useState(false);
   const handleClick = () => {
     setIsModalOpen(true);
   };
 
   const closeModal = () => {
-    console.log("Closing modal", "open? ", isModalOpen);
+    fetchTemplates();
```
Comment (m5matthew) [inline frontend/src/components/CreateTemplateCard/CreateTemplateCard.tsx:18]: nit: is it possible to only re-fetch templates when a user actually creates a new template (eg. on the success handler for the post request)?

Comment (bryanansong) [reply frontend/src/components/CreateTemplateCard/CreateTemplateCard.tsx:18]: Yeah, thanks for the suggestion. I just added this in a new PR
